src/els_client.erl

```diff
@@ -45,6 +46,9 @@
         , start_link/2
         , stop/0
         , workspace_symbol/1
+        , workspace_executecommand/2
+
+        , get_notifications/0
```

Comment:
This function seems only to be used for testing purposes. In that case, I would prefer to not export it and to use this instead:

https://erlang.org/doc/man/sys.html#get_state-1

Reply:
> This function seems only to be used for testing purposes.

What is the purpose of the client?  Either it is for testing, in which case this function makes sense, or it is to actually interact with a server, in which case receiving notifications seems to be an obvious requirement.

What am I missing?

Reply:
> They are provided for convenience, allowing developers to avoid having to create their own state extraction functions

That's the part I was referring to.

Reply:
ok, but the `get_notifications` function returns all notifications received, and empties the list in the state, which `sys:get_state` does not do. This allows a multi-step test to assert on notifications for each step.  This gets more important when you are processing more complex scenarios as lenses and code actions get more capable.